src/facade/dragonfly_connection.cc

```diff
@@ -859,7 +865,7 @@ void Connection::SendAsync(MessageHandle msg) {
   dispatch_q_bytes_.fetch_add(msg.UsedMemory(), memory_order_relaxed);
 
   dispatch_q_.push_back(move(msg));
-  if (dispatch_q_.size() == 1) {
+  if (dispatch_q_.size() == 1 && !cc_->sync_dispatch) {
```

Comment:
there is a case where you push into an empty `dispatch_q_` and do not wake the fiber?

Reply:
Well, we can wake it, but it won't fulfill the event condition: not empty and not sync dispatch

Previously all replies were either sync or all were async. This doesn't hold for multi/exec + subscribe (but its not defined either way), but also for `SUBSCRIBE` if we send the replies synchronously. The change is here so we:

1. Maintain the order in pipelined requests
2. Don't have a "race condition": subscribe needs to update channel store pointers on all threads. While it awaits for all threads to complete the callback, some of them might have done it earlier, to that they already start sending messages before the initial ack is send.

Reply:
I am trying to see if we do not have a deadlock here. I did not deep dive into the exact state machine.
Suppose `dispatch_q_.size() == 1` but `cc_->sync_dispatch` is true. We won't notify.
Now suppose `dispatch_q_.size()` grew to 2 and now `cc_->sync_dispatch` is false. We will never notify.

Reply:
Sync dispatch always notifies after it finished if dispatch q is not empty

Reply:
So if we missed a wake here due to cc_->sync_dispatch, then sync dispatch did preempt, so once it wakes up, it will wake